### ISSUE TITLE
Incorrect diagnostic message for `isinstance(None, object)`

### ISSUE THREAD

**Issue by rkechols** [microsoft/pyright#11155]
**Describe the bug**
Incorrect diagnostic message for `isinstance(None, object)`; this expression evaluates to `True`, but raises the following diagnostic:
```text
Unnecessary isinstance call; "None" is never an instance of "object"  (reportUnnecessaryIsInstance)
```
This message should replace the word `never` with `always` (as it does with `isinstance("hi", object)`, for example)

**Code or Screenshots**
Code sample in [pyright playground](https://pyright-play.net/?pythonVersion=3.12&strict=true&code=A4JwlgdgLgFGDOl5QIYQMYFMYDkD2EmANAAR4BGAVpulAJR0kkDEJAKiAK6ZA)

```python
print(isinstance(None, object))  # True
```

**VS Code extension or command-line**
> Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?

All of the above. I've checked with pyright versions `1.1.407` and `1.1.358`
